### PR TITLE
docs: Added MenuDivider and MenuHeading docs

### PR DESCRIPTION
**Pull request by ElliotTheHuman**
- Added `MenuDivider` and `MenuHeading` doc mentions in Menu docs
- Removed `MenuGroup` doc portion from MenuList docs
- Added note about `density` prop to Menu docs
- Restricted `MenuHeading` children to string (`MenuHeading` has specific styling to match all densities so best to keep this restricted for now, I think)

Before requesting review from another developer please delete the lines above and summarize your change.

### Requirements

Please check the following items are addressed in your pull request (or are not applicable)

- [ ] a11y impact (FUTURE: Make aXe pass req'd for CI ✅)

#### Image snapshots (choose one)

  - [ ] yes
  - [x] not applicable

#### Documentation updated (choose one)

  - [x] yes
  - [ ] not applicable

### CODE REVIEW

File: www/src/documentation/components/overlays/menuitem.mdx [hunk 1]
```diff
@@ -102,19 +102,9 @@ To implement a tooltip on a `MenuItem`, wrap the `MenuItem` in a `Tooltip` eleme
...
 
-**Note:** If a `MenuItem` does not have an icon, but its sibling(s) does have an icon, the `MenuItem` without an icon will automatically allocate space on the left-hand side to align its text up with its sibling(s). This behavior applies to the `MenuItem` children of `MenuList` and `MenuGroup` elements.
-
-```jsx
-<MenuList>
-  <MenuItem icon="LogoRings">Cheddar</MenuItem>
-  <MenuItem>Gouda</MenuItem>
-  <MenuItem icon="ChartPie">Swiss</MenuItem>
-</MenuList>
-```
```
Comment (ElliotTheHuman) [inline www/src/documentation/components/overlays/menuitem.mdx:113]: Icon placeholder behavior is no longer automatically handled by `MenuContext` and is instead provided by the user using the `iconGutter` prop on a parent `Menu` / `MenuList`. This is described in the `Menu` docs.

File: www/src/documentation/components/overlays/menu.mdx [hunk 2]
```diff
@@ -264,3 +296,64 @@ controlled via state values – or use the hook version.
...
+  <Menu
+    content={
+      <>
+        <MenuItem icon="Calendar">Calendar</MenuItem>
+        <MenuItem>No icon</MenuItem>
+        <MenuItem icon="Pivot">Pivot</MenuItem>
+      </>
+    }
+  >
+    <Button>Icon Gutter</Button>
```
Comment (mdodgelooker) [inline www/src/documentation/components/overlays/menu.mdx:317]:
```suggestion
    <Button>No Icon Gutter</Button>
```

File: www/src/documentation/components/overlays/menu.mdx [hunk 3]
```diff
@@ -264,3 +296,64 @@ controlled via state values – or use the hook version.
...
+    iconGutter
+    content={
+      <>
+        <MenuItem icon="Calendar">Calendar</MenuItem>
+        <MenuItem>No icon</MenuItem>
+        <MenuItem icon="Pivot">Pivot</MenuItem>
+      </>
+    }
+  >
+    <Button>No icon Gutter</Button>
```
Comment (mdodgelooker) [inline www/src/documentation/components/overlays/menu.mdx:329]:
```suggestion
    <Button>Icon Gutter</Button>
```

File: www/src/documentation/components/overlays/menulist.mdx [hunk 4]
```diff
@@ -37,36 +46,3 @@ positioning for natural scrolling behavior.
 
 The windowing mode defaults to `fixed`, meaning all children are assumed to have a fixed height.
 This is the most efficient mode, but scrolling will be distorted if the list is grouped.
```
Comment (mdodgelooker) [inline www/src/documentation/components/overlays/menulist.mdx:48]: Might as well delete these too – they're not that helpful now that we don't do groups & variable windowing.
```suggestion
```

**Review by mdodgelooker — CHANGES_REQUESTED** [review]
Looks good! Just a couple minor suggestions.

**Review by mdodgelooker — APPROVED** [review]
LGTM!

**Pull request merged**